pkg/packages/manager.go

```diff
@@ -98,40 +107,43 @@ func processInstalled(mc *ManagerContext) bool {
 		mc.Package.Status.State = api.StateUpdating
 		mc.RequeueAfter = retryShort
 		return true
-	} else {
-		var err error
-		newValues := make(map[string]interface{})
-
-		err = yaml.Unmarshal([]byte(mc.Package.Spec.Config), &newValues)
-		if err != nil {
-			mc.Log.Error(err, "unmarshaling current package configuration")
-			mc.Package.Status.Detail = err.Error()
-			mc.RequeueAfter = retryShort
-			return true
-		}
+	}
+	var err error
+	newValues := make(map[string]interface{})
 
-		newValues[sourceRegistry] = mc.getRegistry(newValues)
-		needs, err := mc.PackageDriver.IsConfigChanged(mc.Ctx, mc.Package.Name,
-			newValues)
-		if err != nil {
-			mc.Log.Error(err, "checking necessity of reconfiguration")
-			mc.Package.Status.Detail = err.Error()
-			mc.RequeueAfter = retryLong
-			return true
-		}
-		if needs {
-			mc.Log.Info("configuration change detected, upgrading")
-			mc.Package.Status.State = api.StateUpdating
-			mc.RequeueAfter = retryShort
-			return true
-		}
+	err = yaml.Unmarshal([]byte(mc.Package.Spec.Config), &newValues)
+	if err != nil {
+		mc.Log.Error(err, "unmarshaling current package configuration")
+		mc.Package.Status.Detail = err.Error()
+		mc.RequeueAfter = retryShort
+		return true
+	}
+
+	newValues[sourceRegistry] = mc.getRegistry(newValues)
+	needs, err := mc.PackageDriver.IsConfigChanged(mc.Ctx, mc.Package.Name, newValues)
+	if err != nil {
+		mc.Log.Error(err, "checking necessity of reconfiguration")
+		mc.Package.Status.Detail = err.Error()
+		mc.RequeueAfter = retryLong
+		return true
+	}
+	if needs {
+		mc.Log.Info("configuration change detected, upgrading")
+		mc.Package.Status.State = api.StateUpdating
+		mc.RequeueAfter = retryShort
+		return true
 	}
 	mc.RequeueAfter = retryVeryLong
 
 	return false
 }
 
 func processUninstalling(mc *ManagerContext) bool {
+	if err := mc.PackageDriver.Initialize(mc.Ctx, mc.getClusterName()); err != nil {
+		mc.Package.Status.Detail = err.Error()
```

Comment:
Does this set the status to error, and is that intended?

Reply:
yes, this allows the user to see the status of an install without looking on controller logs

Reply:
For example, the "Packages namespace..." is the detail here:
```
default                package.packages.eks.amazonaws.com/my-hello   hello-eks-anywhere   119m    unknown                                                      0.1.1-bc7dc6bb874632972cd92a2bca429a846f7aa785 (latest)   Packages namespaces must start with: eksa-packages
````

Reply:
O ok, and this is on the uninstalls